### DISCUSSION TITLE
M5 Client Challenges

### DISCUSSION THREAD

**Discussion by MediaArtsDad** [josephdadams/TallyArbiter#356] [Q&A] [answered]
New to the community here, really appreciate everyone's hard work!

I have used TA for a while now and attempting to add some wireless clients/listeners and running into a few issues.  I am using version 3.0.1 currently.  Following the Pre-Built Installation instructions on each listener client readme.  The .bin files are from workflow run 1/17/21, although I did go back and try a couple of earlier versions with no better luck.

**On the M5Stack ATOM displays:**
1) I am still getting the inversed background colors (red for preview and green for air).  I see elsewhere in discussions this was fixed on the current build, but pulling the .bin from the latest action still appears to have this issue?  

2) The ATOM displays always boot to "1" - if you click the display to change the number, once the assigned device changes in TA, the ATOM display increments to the next number.  For example, after booting, click the display twice, the display shows "2", but once the assigned source changes, the display increments to "3"

**On M5Stick-C**
3) The source text is not displaying.  All the examples I see show the source text (for example, "1 Camera 1" ) showing - am I missing a step in setting this up?

4) If I click on the M5 button, the IP information shows, but clicking back on the M5 button again goes to a blank screen.  The only way to get the Tally color back is to turn off and back on the device.

I would appreciate any help or direction on these!

Thank you
Jeff

**Comment by wrldtvlr** [comment, marked as answer]
For question 1 - I suspect you are using the listener client that was included in release 3.0.1. The change for rgb to grb and a few other things is in the main branch. You should be able to download the latest client code from there. Note - the client code is an Arduino .ino file that needs to be compiled loaded from the Arduino IDE  as I don't think @MatteoGheza and the gang have implemented the auto client update process just yet.

For question 2, the camera number logic is still not quite right but fortunately just cosmetic. The plan will be to send the camera number from the server, but that hasn't been implemented yet.

For question 3 - we don't use a Stick, so I haven't observed that.

For Question 4, I've noticed on the Matrix that it's sometimes necessary to trigger a status transmission from the server to get everything back in sync - i.e. change the Program or Preview camera. You might try that with the Stick.
